common/Geometry2d/PointTest.cpp

```diff
@@ -1,19 +1,219 @@
+    EXPECT_PRED2(pointEqual, expected,
+                 unit.rotated(Point(3, 0), -M_PI / 2 + 12 * M_PI));
+    EXPECT_PRED2(pointEqual, expected,
+                 Point::rotated(unit, Point(3, 0), -M_PI / 2 + 12 * M_PI));
+
+    EXPECT_PRED2(pointEqual, expected.rotated(M_PI / 2), expected.perpCCW());
+    EXPECT_PRED2(pointEqual, expected.rotated(-M_PI / 2), expected.perpCW());
+}
+
 TEST(Point, clampBig) {
     Point p0(10.0, 10.0);
     p0.clamp(10);
     EXPECT_FLOAT_EQ(10, p0.mag());
+    EXPECT_PRED2(pointEqual, p0, Point(sqrt(50), sqrt(50)));
+
+    p0 = Point(-20.5, 1.5);
+    float mag = p0.mag();
+    p0.clamp(1);
+    EXPECT_FLOAT_EQ(1, p0.mag());
+    EXPECT_PRED2(pointEqual, p0, Point(-20.5 / mag, 1.5 / mag));
 }
 
-//  if you clamp a vector to a value that's bigger than it's magnitued, it
-//  shouldn't change
+//  if you clamp a vector to a value that's bigger
+//  than it's magnitued, it shouldn't change
 TEST(Point, clampSmall) {
-    Point p0(10, 0);
+    Point p0(10, -6);
     p0.clamp(15);
-    EXPECT_FLOAT_EQ(10, p0.mag());
+    EXPECT_TRUE(p0 == Point(10, -6));
+    p0.clamp(sqrt(10 * 10 + 6 * 6));
+    EXPECT_PRED2(pointEqual, p0, Point(10, -6));
+}
+
+TEST(Point, distTo) {
+    EXPECT_FLOAT_EQ(0, Point(5, 3).distTo(Point(5, 3)));
+    EXPECT_FLOAT_EQ(5, Point(4, 5).distTo(Point(1, 1)));
+    EXPECT_FLOAT_EQ(5, Point(1.5, 1.5).distTo(Point(4.5, 5.5)));
+    EXPECT_FLOAT_EQ((Point(93, 44) - Point(4.5, 5.5)).mag(),
+                    Point(93, 44).distTo(Point(4.5, 5.5)));
+}
+
+TEST(Point, nearPoint) {
+    for (float y = -5; y <= 5; y += 0.01) {
+        for (float x = -5; x <= 5; x += 0.01) {
+            Point point(x, y);
+            float close = (pow(x - 1.1, 2) + pow(y - 1.3, 2)) - 2.5 * 2.5;
+            if (!floatEqual(close, 0)) {
+                EXPECT_EQ(close <= 0, point.nearPoint(Point(1.1, 1.3), 2.5))
+                    << "x:" << x << " y:" << y;
+            }
+        }
+    }
+}
+
+// TODO(ashaw596) Add tests for the changed normalized
+TEST(Point, normalized) {
+    Point expected(5 / sqrt(5 * 5 + 2 * 2), 2 / sqrt(5 * 5 + 2 * 2));
+    EXPECT_PRED2(pointEqual, expected, Point(5 * 1.4, 2 * 1.4).normalized());
+    EXPECT_PRED2(pointEqual, expected, Point(5 / 1.4, 2 / 1.4).normalized());
+    EXPECT_PRED2(pointEqual, Point(0, 0), Point(0, 0).normalized());
+}
+
+TEST(Point, nearlyEquals) {
+    EXPECT_TRUE(Point(0, 5).nearlyEquals(Point(0, 5)));
+    EXPECT_TRUE(Point(0, 5).nearlyEquals(
+        Point(0 - FLOAT_EPSILON / 2, 5 + FLOAT_EPSILON / 2)));
+    EXPECT_TRUE(Point(0, 5).nearlyEquals(
+        Point(0 + FLOAT_EPSILON / 2, 5 - FLOAT_EPSILON / 2)));
+
+    EXPECT_FALSE(Point(0, 5).nearlyEquals(
+        Point(0 - FLOAT_EPSILON * 1.1, 5 + FLOAT_EPSILON * 1.1)));
+    EXPECT_FALSE(Point(0, 5).nearlyEquals(
+        Point(0 + FLOAT_EPSILON * 1.1, 5 - FLOAT_EPSILON * 1.1)));
 }
+// TODO(ashaw596) Add tests for angleTo and angleBetween once those changes are
```

Comment:
Well, those changes are in my rotation planner stuff. So I can't do this until I merge that branch.